bundles/org.openhab.voice.googlestt/src/main/java/org/openhab/voice/googlestt/internal/GoogleSTTService.java

```diff
@@ -0,0 +1,367 @@
+import org.slf4j.LoggerFactory;
+
+import com.google.api.gax.rpc.ClientStream;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.StreamController;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.cloud.speech.v1.RecognitionConfig;
+import com.google.cloud.speech.v1.SpeechClient;
+import com.google.cloud.speech.v1.SpeechRecognitionAlternative;
+import com.google.cloud.speech.v1.SpeechSettings;
+import com.google.cloud.speech.v1.StreamingRecognitionConfig;
+import com.google.cloud.speech.v1.StreamingRecognizeRequest;
+import com.google.cloud.speech.v1.StreamingRecognizeResponse;
+import com.google.protobuf.ByteString;
+
+import io.grpc.LoadBalancerRegistry;
+import io.grpc.internal.PickFirstLoadBalancerProvider;
+
+/**
+ * The {@link GoogleSTTService} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = SERVICE_PID, property = Constants.SERVICE_PID + "=" + SERVICE_PID)
+@ConfigurableService(category = SERVICE_CATEGORY, label = SERVICE_NAME, description_uri = SERVICE_CATEGORY + ":"
+        + SERVICE_ID)
+public class GoogleSTTService implements STTService {
+
+    private static final String GCP_AUTH_URI = "https://accounts.google.com/o/oauth2/auth";
+    private static final String GCP_TOKEN_URI = "https://accounts.google.com/o/oauth2/token";
+    private static final String GCP_REDIRECT_URI = "urn:ietf:wg:oauth:2.0:oob";
+    private static final String GCP_SCOPE = "https://www.googleapis.com/auth/cloud-platform";
+
+    private final Logger logger = LoggerFactory.getLogger(GoogleSTTService.class);
+    private final ScheduledExecutorService executor = ThreadPoolManager.getScheduledPool("OH-voice-googlestt");
+    private final OAuthFactory oAuthFactory;
+    private final ConfigurationAdmin configAdmin;
+
+    private GoogleSTTConfiguration config = new GoogleSTTConfiguration();
+    private @Nullable OAuthClientService oAuthService;
+
+    @Activate
+    public GoogleSTTService(final @Reference OAuthFactory oAuthFactory,
+            final @Reference ConfigurationAdmin configAdmin) {
+        LoadBalancerRegistry.getDefaultRegistry().register(new PickFirstLoadBalancerProvider());
+        this.oAuthFactory = oAuthFactory;
+        this.configAdmin = configAdmin;
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        var serviceConfig = new org.openhab.core.config.core.Configuration(config).as(GoogleSTTConfiguration.class);
+        this.config = serviceConfig;
+        executor.submit(() -> GoogleSTTLocale.loadLocales(serviceConfig.refreshSupportedLocales));
+        String clientId = serviceConfig.clientId;
+        String clientSecret = serviceConfig.clientSecret;
+        if (!clientId.isEmpty() && !clientSecret.isEmpty()) {
```

Comment:
You can even use isBlank.
In case one of both is not ok, you could log something.